[SC-3670] Clarify --proxy-localhost usage (#1046)
saucelabs/sauce-docs: docs/dev/cli/sauce-connect-proxy.md

### `--proxy-localhost`
<p><small>| OPTIONAL | BOOLEAN |</small></p>

> __Description__: Setting this to `true` supports proxying upstream requests to localhost. This includes the case where an upstream proxy is hosted on localhost, see [Set Up with Additional Proxies](/secure-connections/sauce-connect/setup-configuration/additional-proxies). By default, it is `false`. <br/>

Reviewer: ```suggestion
__Description__: Setting this to `true` supports proxying upstream requests to localhost. This includes scenarios where an [upstream proxy is hosted on localhost](/secure-connections/sauce-connect/setup-configuration/additional-proxies). By default, it is `false`. <br/>
```
